bread/layout/components/tooltip.py

```diff
@@ -0,0 +1,282 @@
+from typing import Optional, Union
+
+import htmlgenerator as hg
+
+from bread.layout.components.button import Button
+from bread.layout.components.icon import Icon
+from bread.utils import Link
+
+
+def _get_icon(icon):
+    """Return a proper Icon instance"""
+    if isinstance(icon, str):
+        icon = Icon(icon, size="16")
+    else:
+        icon.attributes["_class"] = icon.attributes.get("_class", "") + " bx--btn__icon"
+        icon.attributes["width"] = "16"
+        icon.attributes["height"] = "16"
+
+    return icon
+
+
+def _merge_attributes(dest, source):
+    """Bring the class in 'source' inte 'dest'"""
+    if "_class" in source:
+        if source["_class"]:
+            dest["_class"] += source["_class"]
+        del source["_class"]
+
+    # for other attributes, if there exist on both dicts,
+    # just replace them.
+    for attr in dest.keys() & source.keys():
+        dest[attr] = source[attr]
```

Comment:
Could you replace this function with ```htmlgenerator.merge_html_attrs```? We added that some time ago to avoid having to rewrite the same function for every component.

Reply:
OK. Done.